sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AADWebAppConfigurationTest.java

```diff
@@ -7,20 +7,49 @@
 import org.junit.Test;
 import org.springframework.security.oauth2.client.registration.ClientRegistration;
 import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
 
 import java.util.ArrayList;
 import java.util.List;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class AADWebAppConfigurationTest {
 
+    @Test
+    public void noConfigurationOnMissingRequiredProperties() {
+        WebApplicationContextRunnerUtils.getContextRunnerWithoutRequiredProperties()
+            .run(context -> {
+                assertThat(context).doesNotHaveBean("AADWebAppClientRegistrationRepository");
+                assertThat(context).doesNotHaveBean("OAuth2AuthorizedClientRepository");
+                assertThat(context).doesNotHaveBean("OAuth2UserService");
+        });
+    }
+
+    @Test
+    public void configurationOnRequiredProperties() {
+        WebApplicationContextRunnerUtils.getContextRunnerWithRequiredProperties()
+            .run(context -> {
+                ClientRegistrationRepository clientRegistrationRepository = context.getBean(AADWebAppClientRegistrationRepository.class);
+                OAuth2AuthorizedClientRepository authorizedClientRepository = context.getBean(OAuth2AuthorizedClientRepository.class);
+                OAuth2UserService userService = context.getBean(OAuth2UserService.class);
+                assertNotNull(clientRegistrationRepository);
```

Comment:
We can use `assertThat(context).hasSingleBean(AADAuthenticationProperties.class);` instead.